frontend/server/src/main/java/com/amazonaws/ml/mms/http/InferenceRequestHandler.java

```diff
@@ -68,11 +77,33 @@ protected void handleRequest(
                 handlePredictions(ctx, req, segments);
                 break;
             default:
-                handleLegacyPredict(ctx, req, decoder, segments);
+                if (inferEndpointMap.getOrDefault(segments[1], null) != null) {
+                    handleCustomEndpoint(ctx, req, segments);
+                } else {
+                    handleLegacyPredict(ctx, req, decoder, segments);
+                }
                 break;
         }
     }
 
+    private void handleCustomEndpoint(
+            ChannelHandlerContext ctx, FullHttpRequest req, String[] segments) {
+        if (HttpMethod.GET.equals(req.method())) {
+            MMSEndpoint endpoint = inferEndpointMap.get(segments[1]);
+            Runnable r =
+                    () -> {
+                        FullHttpResponse rsp =
+                                new DefaultFullHttpResponse(
+                                        HttpVersion.HTTP_1_1, HttpResponseStatus.OK, false);
+                        endpoint.doGet(req, rsp, new MMSContext());
+                        NettyUtils.sendHttpResponse(ctx, rsp, true);
+                    };
+            ModelManager.getInstance().submitTask(r);
+        } else {
```

Comment:
1. Think about auditing the these runnables. 
2. Handle exceptions thrown endpoints.
3. Look into if this can be scheduled through Netty (execution handler) itself.

Reply:
Will address in the next PR.